### PR TITLE
Use MbedTLS_jll instead of BinaryProvider

### PR DESCRIPTION
**Pull request by JackDunnNZ**
While https://github.com/JuliaLang/julia/pull/32706 is pending, it seems like it would be good to switch this over to the JLL package in the meantime (and the stdlib change won't happen for Julia 1.3 or 1.4).

This is adapted from @quinnj's work in [`jq/stdlib`](https://github.com/JuliaLang/MbedTLS.jl/tree/jq/stdlib) but uses the JLL rather than the Julia-provided library

cc @kmsquire who asked for this here: https://discourse.julialang.org/t/packagecompiler-jl-complaining-that-mbedtls-jl-might-not-be-relocatable/34878

### CODE REVIEW

**Comment by KristofferC** [comment]
Need to update julia compat in Project.toml and update what versions Travis tests.

**Comment by JackDunnNZ** [comment]
Whoops, thanks. I also removed the FORCE_BUILD builds since that's not relevant anymore

File: .travis.yml [hunk 1]
```diff
@@ -1,20 +1,13 @@
 language: julia
 
-sudo: false
-
 os:
   - linux
   - osx
```
Comment (quinnj) [inline .travis.yml]: add `- windows` here

File: appveyor.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 environment:
```
Comment (quinnj) [inline appveyor.yml]: I'd prefer to just get rid of appveyor.yml and test 32-bit/64-bit windows via travis. It's nice to have all the OS and platforms in a single CI run.

File: .travis.yml (hunk 1 above)
Comment (JackDunnNZ) [reply .travis.yml]: Done

File: appveyor.yml (hunk 2 above)
Comment (JackDunnNZ) [reply appveyor.yml]: Makes sense, removed

**Comment by kmsquire** [comment]
I'll add my thanks!  Can someone please tag a release?

**Pull request merged**